File: client-clojure/src/sirqul_io_t_platform/specs/service_hub.clj
(ns sirqul-io-t-platform.specs.service-hub
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.retailer :refer :all]
            [sirqul-io-t-platform.specs.location :refer :all]
            )
  (:import (java.io File)))


(def service-hub-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :locationDescription) string?
   (ds/opt :name) string?
   (ds/opt :externalId) string?
   (ds/opt :business) retailer-spec
   (ds/opt :location) location-spec
   (ds/opt :timeZone) string?
   })

(def service-hub-spec
  (ds/spec
    {:name ::service-hub
     :spec service-hub-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ServiceHub.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServiceHub._

case class ServiceHub (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
latitude: Option[Double],
longitude: Option[Double],
locationDescription: Option[String],
name: Option[String],
externalId: Option[String],
business: Option[Retailer],
location: Option[Location],
timeZone: Option[String])

object ServiceHub {
  import DateTimeCodecs._

  implicit val ServiceHubCodecJson: CodecJson[ServiceHub] = CodecJson.derive[ServiceHub]
  implicit val ServiceHubDecoder: EntityDecoder[ServiceHub] = jsonOf[ServiceHub]
  implicit val ServiceHubEncoder: EntityEncoder[ServiceHub] = jsonEncoderOf[ServiceHub]
}
